### DISCUSSION TITLE
Start polling on HX Trigger header

### DISCUSSION THREAD

**Discussion by vietk** [bigskysoftware/htmx#2543] [Ideas]
Hello

I don't know if it's possible to do that with HTMX, I spent some time searching about it but could not find anything.

I have a div element that is doing hx-get with a polling interval, however I want to trigger this polling after having launched a background job on server side.
I think it would be nice that the polling condition could be started when the server send back a Hx-Trigger header response.

```html
<form ... hx-post="/startBackground" hx-swap="none"> ...
<div hx-get="/get/task/1" hx-trigger=every 1s [polling]> 
```

Regards

**Comment by Nanimo1309** [comment]
If I get it right, you can create 
```
<form hx-post="/startBackground" hx-swap="afterend">
```
this can start process and return
```
<div hx-get="/get/task/1" hx-trigger="every 1s">
```
this will be appended after form and start polling

**Reply by vietk** [reply]
Hello ! 
Thanks for the response, this is, more or less what I am doing: 

I am using a fragment with a div + a target to the fragement from the form element.

```
<form ... hx-post="/startBackground" hx-target="polling"> ...
<div id="polling" hx-get="/get/task/1" hx-trigger="every 1s"> 
```

The idea with the event is to avoid the usage of an extra fragment or a div and start polling from the existing element.

Regards

**Reply by Nanimo1309** [reply]
While it is good to show that your process is running via another div with information, you can try
```
<form ... hx-trigger="submit, notDone delay:1s">
```
and send HX-Trigger: notDone HTTP header, but you must have in mind that this will send whole form every time

**Comment by vietk** [comment]
ok not really what I have in mind, I will stick the proposed "polling" div
workaround, it took me some time to reason about but it works perfectly

Tanks for your answers and your time !
